src/ImageSharp/ApplyProcessors.cs

```diff
@@ -0,0 +1,107 @@
+﻿// <copyright file="RotateFlip.cs" company="James Jackson-South">
+// Copyright (c) James Jackson-South and contributors.
+// Licensed under the Apache License, Version 2.0.
+// </copyright>
+
+namespace ImageSharp
+{
+    using System;
+
+    using ImageSharp.PixelFormats;
+
+    using ImageSharp.Processing;
+
+    /// <summary>
+    /// Extension methods for the <see cref="Image{TPixel}"/> type.
+    /// </summary>
+    public static partial class ImageExtensions
+    {
+        /// <summary>
+        /// Mutates the image by applying the operations to it.
+        /// </summary>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        /// <param name="source">The image to rotate, flip, or both.</param>
+        /// <param name="operations">The operations to perform on the source.</param>
+        public static void Mutate<TPixel>(this Image<TPixel> source, Action<IImageOperations<TPixel>> operations)
```

Comment:
Bounds() should be an extension method I feel, as its only really used as a helper for image operations. (in fact it should probably just be internal)

We should also probably move out all all the overloads of `Save()` into extension methods too `Image` would just have `Save(Stream, Encoder)`.

Also things like `To<TPixel2>()` could also probably be a candidate for changing. 

And now the `ApplyProcessor()` is basically redundant.